options.go

```diff
@@ -195,6 +195,15 @@ func Runtime(r runtime.Runtime) Option {
 	}
 }
 
+// Router sets the router
+func Router(r router.Router) Option {
+	return func(o *Options) {
+		o.Router = r
+		// Update client
+		o.Client.Init(client.Router(r))
```

Comment:
Do we need to do nil checking here or is `o.Client` always non nil?

Reply:
It's always non-nil